app/src/main/java/com/poupa/vinylmusicplayer/discog/Discography.java

```diff
@@ -124,6 +124,33 @@ public Song getSong(long songId) {
         }
     }
 
+    @NonNull
+    public ArrayList<Song> getSongsFromIdsAndCleanupOrphans(@NonNull ArrayList<Long> songIds, @Nullable Consumer<ArrayList<Long>> orphanIdsCleaner) {
+        ArrayList<Long> orphanSongIds = new ArrayList<>();
+        ArrayList<Song> songs = new ArrayList<>();
+
+        synchronized (cache) {
+            for (Long id : songIds) {
+                Song song = cache.songsById.get(id);
+                if (song != null) {
+                    songs.add(song);
+                } else if (orphanIdsCleaner != null) {
+                    orphanSongIds.add(id);
+                }
+            }
+        }
+
+        MemCache.ConsistencyState cacheState = getCacheState();
+        // In the case where the Discog is being reset situation, the operation takes time,
+        // and while the cache is being filled up,
+        // correct and existing songs may be considered as orphan
+        // --> incorrectly cleaned from the auxiliary DBs (history, queue, etc)
+        if ((orphanIdsCleaner != null) && (cacheState != MemCache.ConsistencyState.RESETTING)) {
```

Comment:
I know it should be very unlikely as reset is only activated when the user in clicking on the rescan library button but is more of a what if the reset state is used for another thing later on ? Because the bug I have introduce seems to enable crash on boot that can only be stop by emptying the cache/memory of the app.

Reply:
> Could this make that we can miss real orphans songs ? If cache resetting is done at the same time that for example the queue is restoring

Possible, but (kind of) tolerable since we will just not cleanup the orphan. For entries DB, the entries gets cleaned up on the next call to fetch the playlist (quite often). For playing queue, till the next launch of the app.

Reply:
> I know it should be very unlikely as reset is only activated when the user in clicking on the rescan library button but is more of a what if the reset state is used for another thing later on ? Because the bug I have introduce seems to enable crash on boot that can only be stop by emptying the cache/memory of the app.

I think it's impossible, or very unlikely, to interact with UI to trigger reset while the MusicService class is being initialized (queue loading etc).

You can avoid emptying cache by applying this patch (I hate loosing play history), or surround the queue loading logic with try/catch (the previous version of the app has this behaviour if my memory is correct, the queue is sacrified to be able to start the app)

Reply:
👍